beacon-chain/blockchain/forkchoice_update_execution.go

```diff
@@ -40,28 +42,57 @@ func (s *Service) getStateAndBlock(ctx context.Context, r [32]byte) (state.Beaco
 	return headState, newHeadBlock, nil
 }
 
-func (s *Service) forkchoiceUpdateWithExecution(ctx context.Context, newHeadRoot [32]byte) error {
+func (s *Service) forkchoiceUpdateWithExecution(ctx context.Context, newHeadRoot [32]byte, proposingSlot primitives.Slot) error {
 	isNewHead := s.isNewHead(newHeadRoot)
-	if !isNewHead && !s.isNewProposer() {
+	if !isNewHead && !s.isNewProposer(proposingSlot) {
 		return nil
 	}
 
-	headState, headBlock, err := s.getStateAndBlock(ctx, newHeadRoot)
-	if err != nil {
-		log.WithError(err).Error("Could not get forkchoice update argument")
-		return nil
+	var headState state.BeaconState
+	var headBlock interfaces.ReadOnlySignedBeaconBlock
+	var headRoot [32]byte
+	var err error
+
+	shouldUpdate := isNewHead
+	if isNewHead && s.isNewProposer(proposingSlot) && !features.Get().DisableReorgLateBlocks {
+		if proposingSlot == s.CurrentSlot() {
```

Comment:
I think this needs to be `s.CurrentSlot() + 1`

Reply:
Nope, this function may be called by the proposer right before proposing at second zero, this is the logic in that case.